modules/afsocket/afsocket-dest.c

```diff
@@ -312,11 +312,13 @@ afsocket_dd_setup_addresses_method(AFSocketDestDriver *self)
 }
 
 static void
-afsocket_dd_restore_connection(AFSocketDestDriver *self)
+afsocket_dd_restore_writer(AFSocketDestDriver *self)
 {
   GlobalConfig *cfg = log_pipe_get_config(&self->super.super.super);
-
-  self->writer = cfg_persist_config_fetch(cfg, afsocket_dd_format_persist_name(self, FALSE));
+  LogWriter *fetched_writer = cfg_persist_config_fetch(cfg, afsocket_dd_format_persist_name(self, FALSE));
```

Comment:
self->writer should be NULL at this phase.
So assert that presumption, and set the self->writer directly